### PR TITLE
drivers: adc: Add drivers/adc to private include

### PR DESCRIPTION
**Pull request by kurddt**
This will allow to use `adc_context.h` from out-of-tree
drivers added to the library using `zephyr_library_amend`

Signed-off-by: Guillaume Lager <g.lager@innoseis.com>

### CODE REVIEW

**Review by galak — CHANGES_REQUESTED** [review]
We don't do this elsewhere.  Not sure if its a general issue for supporting drivers out of tree or not.

@tejlmand is the right solution and do we need to make this change elsewhere?

**Comment by kurddt** [comment]
@galak I'm not sure that every drivers have "private" common include file like `adc_context.h`. if not there is no need to have the driver directory in the include path

**Comment by galak** [comment]
> @galak I'm not sure that every driver have "private" common include file like `adc_context.h`. if not there is no need to have the driver directory in the include path

Not ever driver does, but a number of other driver classes have private headers.  So we should have a common solution and make this work for those cases.  For example `spi`, `eth`, `modem`, `led`

**Comment by kurddt** [comment]
I can make a separate PR that add the library directory in the private include directories when calling `zephyr_library()`

File: drivers/adc/CMakeLists.txt [hunk 1]
```diff
@@ -20,3 +20,4 @@ zephyr_library_sources_ifdef(CONFIG_ADC_NPCX		adc_npcx.c)
 zephyr_library_sources_ifdef(CONFIG_USERSPACE		adc_handlers.c)
 zephyr_library_sources_ifdef(CONFIG_ADC_CC32XX		adc_cc32xx.c)
 zephyr_library_sources_ifdef(CONFIG_ADC_EMUL		adc_emul.c)
+zephyr_library_include_directories(${CMAKE_CURRENT_LIST_DIR})
```
Comment (tejlmand) [inline drivers/adc/CMakeLists.txt:23]: Relative paths are relative to `CMAKE_CURRENT_SOURCE_DIR`, so you can also do.
```
zephyr_library_include_directories(.)
```

**Review by tejlmand — APPROVED** [review]
LGTM.

Consider to use just a `.` instead.

**Comment by tejlmand** [comment]
> We don't do this elsewhere. Not sure if its a general issue for supporting drivers out of tree or not.
> 
> @tejlmand is the right solution and do we need to make this change elsewhere?

We do it here:
https://github.com/zephyrproject-rtos/zephyr/blob/bfd271932228f8ce33b68b82ffee5ee3b2386a17/subsys/net/ip/CMakeLists.txt#L4

and we do this many places:
https://github.com/zephyrproject-rtos/zephyr/blob/bfd271932228f8ce33b68b82ffee5ee3b2386a17/subsys/canbus/canopen/CMakeLists.txt#L9
although that's a bit different as that exports the include path to Zephyr interface, which in return makes it available to everyone.

File: drivers/adc/CMakeLists.txt (hunk 1 above)
Comment (kurddt) [reply drivers/adc/CMakeLists.txt]: done

**Review by tejlmand — CHANGES_REQUESTED** [review]
> This will allow to use adc_context.h from out-of-tree
drivers added to the library using zephyr_library_amend

Actually, before having changes upstream, could you try the following in you downstream code to see if that solves your problem of the include path:
```
zephyr_library_amend()
zephyr_library_include_directories($<TARGET_PROPERTY:SOURCE_DIR>)
.... rest of your code ...
```

**Comment by kurddt** [comment]
@tejlmand Yes it does

**Comment by tejlmand** [comment]
> Should we indicate this solution in the `zephyr_library_amend` documentation instead ?

I think that would be nice, although we should be careful not to describe all features of CMake, something like this is indeed useful.

**Comment by tejlmand** [comment]
> > Should we indicate this solution in the `zephyr_library_amend` documentation instead ?
> 
> I think that would be nice, although we should be careful not to describe all features of CMake, something like this is indeed useful.

Done, see: #35869

**Comment by kurddt** [comment]
Thanks I will close my pull request then

**Pull request closed without merging**